politeiawww/backend.go

```diff
@@ -86,6 +87,8 @@ type backend struct {
 
 	// Following entries require locks
 	userPubkeys     map[string]string               // [pubkey][userid]
+	userEmail       map[string]string               // [email][userid]
+	userUsername    map[string]string               // [username][userid]
```

Comment:
I don't think we should be adding in-memory caches to www if we're just going to have to remove them in a few weeks.  The way I suggest doing this would be to store the user record as:

```
type User struct {
	UserID   string // primary key
	Username string 
	Payload  []byte // encrypted blob with the rest of the user details
}
```

That way we can do user lookups using either the userID or username and eliminate the need for a `userUsername` in-memory cache.  The `userEmail` in-memory cache would still be required temporarily, but since we want to make email optional anyway, we'll need to switch all email lookups in www to be either username or userID lookups.  When we do this we can remove the `userEmail` in-memory cache completely.